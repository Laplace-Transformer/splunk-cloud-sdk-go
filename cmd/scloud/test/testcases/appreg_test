appreg create-app-native --name scloud_test_appreg_name1 --title scloud_test_appreg_title1 --description scloud_test_appreg_description --redirect-urls https://scloud.test.appreg.redirect1.url.com,https://scloud.test.appreg.redirect2.url.com --login-url https://scloud.test.appreg.login.url.com --logo-url https://scloud.test.appreg.logo.url.com --setup-url https://scloud.test.appreg.setup.url.com --user-permissions-filter scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4 --app-principal-permissions scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4
appreg create-app-service --name scloud_test_appreg_name2 --title scloud_test_appreg_title2 --description scloud_test_appreg_description --redirect-urls https://scloud.test.appreg.redirect1.url.com,https://scloud.test.appreg.redirect2.url.com --login-url https://scloud.test.appreg.login.url.com --logo-url https://scloud.test.appreg.logo.url.com --setup-url https://scloud.test.appreg.setup.url.com --user-permissions-filter scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4 --app-principal-permissions scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4

appreg update-app --app-name scloud_test_appreg_name --title scloud_test_appreg_title_updated --description scloud_test_appreg_description_updated --redirect-urls https://scloud.test.appreg.redirect.updated.url.com,https://scloud.test.appreg.redirect2.updated.url.com --login-url https://scloud.test.appreg.login.updated.url.com --logo-url https://scloud.test.appreg.logo.updated.url.com --setup-url https://scloud.test.appreg.setup.updated.url.com --user-permissions-filter scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4 --app-principal-permissions scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4

appreg get-app --app-name scloud_test_appreg_name
appreg list-apps

appreg create-subscription --app-name scloud_test_appreg_name
appreg get-subscription --app-name scloud_test_appreg_name
appreg list-app-subscriptions --app-name scloud_test_appreg_name
appreg list-subscriptions
appreg delete-subscription --app-name scloud_test_appreg_name

appreg get-keys
appreg rotate-secret --app-name scloud_test_appreg_name

appreg delete-app --app-name scloud_test_appreg_name

