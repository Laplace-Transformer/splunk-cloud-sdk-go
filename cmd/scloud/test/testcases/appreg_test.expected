#testcase: appreg create-app-web-app --name scloud_test_appreg_name --kind web --title scloud_test_appreg_title --description scloud_test_appreg_description --redirect-urls https://scloud.test.appreg.redirect1.url.com,https://scloud.test.appreg.redirect2.url.com --login-url https://scloud.test.appreg.login.url.com --logo-url https://scloud.test.appreg.logo.url.com --setup-url https://scloud.test.appreg.setup.url.com --user-permissions-filter scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4 --app-principal-permissions scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4
REQUEST URL:app-registry/v1beta2/apps
REQUEST BODY:{{"kind":"web","name":"scloud_test_appreg_name","redirectUrls":["https://scloud.test.appreg.redirect1.url.com","https://scloud.test.appreg.redirect2.url.com"],"title":"scloud_test_appreg_title","appPrincipalPermissions":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"],"description":"scloud_test_appreg_description","loginUrl":"https://scloud.test.appreg.login.url.com","logoUrl":"https://scloud.test.appreg.logo.url.com","setupUrl":"https://scloud.test.appreg.setup.url.com","userPermissionsFilter":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"]}}

#testcase: appreg create-app-native-app --name scloud_test_appreg_name1 --kind native --title scloud_test_appreg_title1 --description scloud_test_appreg_description --redirect-urls https://scloud.test.appreg.redirect1.url.com,https://scloud.test.appreg.redirect2.url.com --login-url https://scloud.test.appreg.login.url.com --logo-url https://scloud.test.appreg.logo.url.com --setup-url https://scloud.test.appreg.setup.url.com --user-permissions-filter scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4 --app-principal-permissions scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4
REQUEST URL:app-registry/v1beta2/apps
REQUEST BODY:{{"kind":"native","name":"scloud_test_appreg_name1","title":"scloud_test_appreg_title1","appPrincipalPermissions":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"],"description":"scloud_test_appreg_description","loginUrl":"https://scloud.test.appreg.login.url.com","logoUrl":"https://scloud.test.appreg.logo.url.com","redirectUrls":["https://scloud.test.appreg.redirect1.url.com","https://scloud.test.appreg.redirect2.url.com"],"setupUrl":"https://scloud.test.appreg.setup.url.com","userPermissionsFilter":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"]}}

#testcase: appreg create-app-service-app --name scloud_test_appreg_name2 --kind service --title scloud_test_appreg_title2 --description scloud_test_appreg_description --redirect-urls https://scloud.test.appreg.redirect1.url.com,https://scloud.test.appreg.redirect2.url.com --login-url https://scloud.test.appreg.login.url.com --logo-url https://scloud.test.appreg.logo.url.com --setup-url https://scloud.test.appreg.setup.url.com --user-permissions-filter scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4 --app-principal-permissions scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4
REQUEST URL:app-registry/v1beta2/apps
REQUEST BODY:{{"kind":"service","name":"scloud_test_appreg_name2","title":"scloud_test_appreg_title2","appPrincipalPermissions":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"],"description":"scloud_test_appreg_description","loginUrl":"https://scloud.test.appreg.login.url.com","logoUrl":"https://scloud.test.appreg.logo.url.com","redirectUrls":["https://scloud.test.appreg.redirect1.url.com","https://scloud.test.appreg.redirect2.url.com"],"setupUrl":"https://scloud.test.appreg.setup.url.com","userPermissionsFilter":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"]}}

#testcase: appreg update-app --app-name scloud_test_appreg_name --title scloud_test_appreg_title_updated --description scloud_test_appreg_description_updated --redirect-urls https://scloud.test.appreg.redirect.updated.url.com,https://scloud.test.appreg.redirect2.updated.url.com --login-url https://scloud.test.appreg.login.updated.url.com --logo-url https://scloud.test.appreg.logo.updated.url.com --setup-url https://scloud.test.appreg.setup.updated.url.com --user-permissions-filter scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4 --app-principal-permissions scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2,scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4
REQUEST URL:app-registry/v1beta2/apps/scloud_test_appreg_name
REQUEST BODY:{{"appPrincipalPermissions":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"],"description":"scloud_test_appreg_description_updated","loginUrl":"https://scloud.test.appreg.login.updated.url.com","logoUrl":"https://scloud.test.appreg.logo.updated.url.com","redirectUrls":["https://scloud.test.appreg.redirect.updated.url.com","https://scloud.test.appreg.redirect2.updated.url.com"],"setupUrl":"https://scloud.test.appreg.setup.updated.url.com","title":"scloud_test_appreg_title_updated","userPermissionsFilter":["scloud_test_appreg_resource1:scloud_test_appreg_scope1.scloud_test_appreg_scope2","scloud_test_appreg_resource2:scloud_test_appreg_scope3.scloud_test_appreg_scope4"]}}

#testcase: appreg get-app --app-name scloud_test_appreg_name
REQUEST URL:app-registry/v1beta2/apps/scloud_test_appreg_name
REQUEST BODY:<nil>

#testcase: appreg list-apps
REQUEST URL:app-registry/v1beta2/apps
REQUEST BODY:<nil>

#testcase: appreg create-subscription --app-name scloud_test_appreg_name
REQUEST URL:app-registry/v1beta2/subscriptions
REQUEST BODY:{{"appName":"scloud_test_appreg_name"}}

#testcase: appreg get-subscription --app-name scloud_test_appreg_name
REQUEST URL:app-registry/v1beta2/subscriptions/scloud_test_appreg_name
REQUEST BODY:<nil>

#testcase: appreg list-app-subscriptions --app-name scloud_test_appreg_name
REQUEST URL:app-registry/v1beta2/apps/scloud_test_appreg_name/subscriptions
REQUEST BODY:<nil>

#testcase: appreg list-subscriptions
REQUEST URL:app-registry/v1beta2/subscriptions
REQUEST BODY:<nil>

#testcase: appreg delete-subscription --app-name scloud_test_appreg_name
REQUEST URL:app-registry/v1beta2/subscriptions/scloud_test_appreg_name
REQUEST BODY:<nil>

#testcase: appreg get-keys
REQUEST URL:app-registry/v1beta2/keys
REQUEST BODY:<nil>

#testcase: appreg rotate-secret --app-name scloud_test_appreg_name
REQUEST URL:app-registry/v1beta2/apps/scloud_test_appreg_name/rotate-secret
REQUEST BODY:<nil>

#testcase: appreg delete-app --app-name scloud_test_appreg_name
REQUEST URL:app-registry/v1beta2/apps/scloud_test_appreg_name
REQUEST BODY:<nil>

